packages/nodes-base/nodes/Hubspot/Hubspot.node.ts

```diff
@@ -935,8 +1026,206 @@ export class Hubspot implements INodeType {
+						const readRequestBody = {
+							idProperty: idProperty || undefined,
+							inputs: ([...new Array(batchSize)]).map((_, index) => ({ id: this.getNodeParameter('objectId', index + batchStart) as string })),
+						};
+						const readEndpoint = `/crm/v3/objects/${customObjectType}/batch/read`;
+						const readResponse = await hubspotApiRequest.call(this, 'POST', readEndpoint, readRequestBody);
+						readResponse.results.forEach((result: { id: string, properties: Record<string, string> }) => {
+							idMap[idProperty ? result.properties[idProperty] : result.id] = result.id;
+						});
+
+
+						const getProperties = (index: number) => {
+							const additionalFields = this.getNodeParameter('additionalFields', index) as IDataObject;
+							const objectId = this.getNodeParameter('objectId', index) as IDataObject;
+							const properties: Record<string, unknown> = idProperty ? {
+								[idProperty]: objectId,
+							} : {};
+							if (additionalFields.customPropertiesUi) {
+								const customProperties = (additionalFields.customPropertiesUi as IDataObject).customPropertiesValues as IDataObject[];
+
+								if (customProperties) {
+									for (const customProperty of customProperties) {
+										properties[customProperty.property as string] = customProperty.value;
+									}
+								}
+							}
+							return properties;
+						};
+
+						const updateBody = {
+							inputs: [...new Array(batchSize)].map((_, index) =>
+							({
+								id: idMap[this.getNodeParameter('objectId', index + batchStart) as string],
+								properties: getProperties(index + batchStart),
+							}),
+							).filter(element => element.id !== undefined),
+						};
+
+						const createBody = {
+							inputs: [...new Array(batchSize)].map((_, index) =>
+							({
+								id: idMap[this.getNodeParameter('objectId', index + batchStart) as string],
+								properties: getProperties(index + batchStart),
+							}),
+							)
+								.filter(element => element.id === undefined)
+								.map(element => ({ properties: element.properties })),
+						};
+
+						const createEndpoint = `/crm/v3/objects/${customObjectType}/batch/create`;
+						const updateEndpoint = `/crm/v3/objects/${customObjectType}/batch/update`;
+
+						const createResponse = createBody.inputs.length > 0
+							? await hubspotApiRequest.call(this, 'POST', createEndpoint, createBody)
+							: undefined;
+
+						const updateResponse = updateBody.inputs.length > 0
+							? await hubspotApiRequest.call(this, 'POST', updateEndpoint, updateBody)
+							: undefined;
+
+						errors.push(...(createResponse ? getErrorsFromBatchResponse(createResponse) : []));
+						errors.push(...(updateResponse ? getErrorsFromBatchResponse(updateResponse) : []));
+
+						return [...(createResponse?.results || []), ...(updateResponse?.results || []), ...errors];
+					}
+					return [];
+
+				} catch (error) {
+					if (this.continueOnFail()) {
+						if (resource === 'customObject') {
```

Comment:
The first check is just there, because the batch operations don't behave in the expected way, if continue on fail is not set. It could basically be removed and the code would still work.

This check checks it again because here it is actually wrong to parse additional information and return it if we don't `continueOnFail`. It cannot be removed.